// masonry-component/esm2020/lib/masonry-component.component.mjs
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
export class MasonryComponentComponent {
    constructor() {
        this.images = [{
                caption: "first note",
                imageUrl: "./assets/Image_1.jpg"
            }];
    }
    ngOnInit() {
    }
}
MasonryComponentComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MasonryComponentComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: MasonryComponentComponent, selector: "lib-masonry-component", ngImport: i0, template: `
    <div class="masonry">
      <div class="mItem" *ngFor="let img of images">
        <img src="{{img.imageUrl}}" alt="{{img.caption}}">
      </div>
    </div>
  `, isInline: true, styles: ["img{width:100%}.masonry{column-count:4;grid-column-gap:16px;column-gap:16px}.masonry .mItem{display:inline-block;margin-bottom:16px;width:100%}@media (max-width: 1199px){.masonry{column-count:3}}@media (max-width: 991px){.masonry{column-count:2}}@media (max-width: 767px){.masonry{column-count:1}}\n"] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'lib-masonry-component',
                    template: `
    <div class="masonry">
      <div class="mItem" *ngFor="let img of images">
        <img src="{{img.imageUrl}}" alt="{{img.caption}}">
      </div>
    </div>
  `,
                    styles: ['img {width: 100%;}.masonry {column-count: 4;column-gap: 16px;}.masonry .mItem {display: inline-block;margin-bottom: 16px;width: 100%;}@media (max-width: 1199px) {.masonry {column-count: 3;}}@media (max-width: 991px) {.masonry {column-count: 2;}}@media (max-width: 767px) {.masonry {column-count: 1;}}']
                }]
        }], ctorParameters: function () { return []; } });
