// masonry-component/esm2020/lib/masonry-component.module.mjs
import { NgModule } from '@angular/core';
import { MasonryComponentComponent } from './masonry-component.component';
import * as i0 from "@angular/core";
export class MasonryComponentModule {
}
MasonryComponentModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MasonryComponentModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentModule, declarations: [MasonryComponentComponent], exports: [MasonryComponentComponent] });
MasonryComponentModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        MasonryComponentComponent
                    ],
                    imports: [],
                    exports: [
                        MasonryComponentComponent
                    ]
                }]
        }] });
