// masonry-component/fesm2020/masonry-component.mjs
import * as i0 from '@angular/core';
import { Injectable, Component, NgModule } from '@angular/core';

class MasonryComponentService {
    constructor() { }
}
MasonryComponentService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
MasonryComponentService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return []; } });

class MasonryComponentComponent {
    constructor() {
        this.images = [{
                caption: "first note",
                imageUrl: "./assets/Image_1.jpg"
            }];
    }
    ngOnInit() {
    }
}
MasonryComponentComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MasonryComponentComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: MasonryComponentComponent, selector: "lib-masonry-component", ngImport: i0, template: `
    <div class="masonry">
      <div class="mItem" *ngFor="let img of images">
        <img src="{{img.imageUrl}}" alt="{{img.caption}}">
      </div>
    </div>
  `, isInline: true, styles: ["img{width:100%}.masonry{column-count:4;grid-column-gap:16px;column-gap:16px}.masonry .mItem{display:inline-block;margin-bottom:16px;width:100%}@media (max-width: 1199px){.masonry{column-count:3}}@media (max-width: 991px){.masonry{column-count:2}}@media (max-width: 767px){.masonry{column-count:1}}\n"] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'lib-masonry-component',
                    template: `
    <div class="masonry">
      <div class="mItem" *ngFor="let img of images">
        <img src="{{img.imageUrl}}" alt="{{img.caption}}">
      </div>
    </div>
  `,
                    styles: ['img {width: 100%;}.masonry {column-count: 4;column-gap: 16px;}.masonry .mItem {display: inline-block;margin-bottom: 16px;width: 100%;}@media (max-width: 1199px) {.masonry {column-count: 3;}}@media (max-width: 991px) {.masonry {column-count: 2;}}@media (max-width: 767px) {.masonry {column-count: 1;}}']
                }]
        }], ctorParameters: function () { return []; } });

class MasonryComponentModule {
}
MasonryComponentModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MasonryComponentModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentModule, declarations: [MasonryComponentComponent], exports: [MasonryComponentComponent] });
MasonryComponentModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: MasonryComponentModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        MasonryComponentComponent
                    ],
                    imports: [],
                    exports: [
                        MasonryComponentComponent
                    ]
                }]
        }] });

/*
 * Public API Surface of masonry-component
 */

/**
 * Generated bundle index. Do not edit.
 */

export { MasonryComponentComponent, MasonryComponentModule, MasonryComponentService };
